config/config.go

```diff
@@ -0,0 +1,14 @@
+package config
+
+// Config represents a parsed datafile config.
+type Config interface {
```

Comment:
idea is the keep the interfaces/API as similar the original JS SDK: https://github.com/featurevisor/featurevisor/tree/main/packages/sdk/src

this is how we are approaching Swift and Kotlin SDKs as well, helping keep it relatively easy to port bug fixes everywhere easily continuously. 

- https://github.com/featurevisor/featurevisor-swift
- https://github.com/featurevisor/featurevisor-kotlin